### PR TITLE
KoGPT2 베이스라인 코드

### PR DESCRIPTION
**Pull request by ktde24**
1. 전처리 과정
- 필수 필드(caption, name, i_action, classification)는 항상 포함하도록
- 선택 필드(character, setting, action 등)는 값이 없을 경우 ```<empty>```로 처리
- 데이터의 일관성을 유지하기 위해 랜덤 시드 설정 및 선택 필드 순서 섞기 적용

2. Pre-trained Model 적용
- Hugging Face의 skt/kogpt2-base-v2를 사용하여 모델 및 토크나이저 로드
- 새로운 데이터를 더 잘 이해할 수 있도록 특수 토큰 추가 및 임베딩 크기 조정

3. 훈련
- 옵티마이저로 ```AdamW``` 사용, 손실 함수로 ```CrossEntropyLoss``` 적용

4. 현재 문제점
- 모델이 훈련 후에도 입력 필드만 그대로 출력하는 현상 발생
=> 데이터 수가 적어서인지, hyperparameter 튜닝 문제인지, 새로 만들어준 토큰을 단순한
텍스트로 학습한 것인지는 파악 필요...

### CODE REVIEW

**Comment by chasubeen** [comment]
정아님 코드 확인하였고, 저와 비슷한 문제가 발생하고 있는 것 같아요..! 
제 생각에는 다음 부분들을 검토해 봐야 할 것 같긴 합니다.

## 1. model 가중치 load
- koGPT 성능을 측정하기 위해 진행된 task에 맞춰 모델 추론을 돌렸을 때는 정상적인 결과가 나오는가
(ex. 문서 요약>> 허깅페이스에서 모델 불러와서 `model.generate()` 하였을 때 제대로 된 문서 요약 결과가 출력되는가)
- 가끔 허깅페이스에 올라와있는 모델 가중치 자체가 문제인 경우들도 좀 있어 확인은 한 번 해보는 게 나을 것 같긴 해요..ㅎㅎ
- 추가적으로 pre-trained tokenizer/model 등을 가져올 때 'vocabulary map 등도 같이 가져와야한다' 등 같은.. 요구되는 추가 작업들이 있는지도 확인해야될 것 같습니다. 
(README 재확인 후 필요한 설정들은 업데이트 해야할 것 같아요 ㅜ)

## 2. 추론 과정
### 2-1. 학습량 부족  
  - 확실히 훈련 후에도 입력 필드를 그대로 출력하는 듯한 추론 결과가 도출되고 있긴 합니다.
    - 이 부분은 저희가 sample로 활용한 데이터의 건수가 매우 작은 점으로 미루어 보아, 어쩌면 당연한 결과일 수도 있을 것 같습니다.  
    (매우 작은 데이터셋으로 학습된 결과(가중치)로 모델이 업데이트 된 것이니..?)
    - 이후 튜닝 시의 파라미터 효율성 + 원본 파라미터를 최대한 보존하는 방향으로 진행하려면 PEFT(+LoRA) 등을 활용하여 일부 레이어만 튜닝하는 전략을 택할 수도 있을 듯 합니다.

### 2-2. 특수 토큰
- 현재 특수 토큰의 경우 tokenizer에는 제대로 추가되고 있는 것 같은데(json 파일 등도 한 번 열어서 확인은 해보시는 게 좋을 것 같긴 해요!), 모델 학습 및 추론 시에도 반영이 되고 있는건가는 조금 의심스럽긴 합니다.
  - attention mask 쪽에 해당 작업을 딱히 조건화하지는 않아서, masking이 제대로 걸리는 상황인지에 대한 점검은 필요해 보입니다.  
  (token의 경우는 학습 대상이 아님을 모델에 명시해야 할 것 같음, 그러나 현재 해당 부분이 반영되고 있는 것인가는 의심됨)

## 3. 하이퍼 파라미터 조정
- README 뒤져보면 파인 튜닝 시에는 어떠한 하이퍼 파라미터 조합으로 튜닝해라/ 또는 지난 번에 참고자료로 찾아오신 프로젝트들을 보면 활용하고 있는 파라미터 조합들을 확인할 수는 있을 것 같아요.
- 해당 조합들 확인 후 변경해서 시도해 보아야 할 것 같아요.
